sensor/common/registry/registry_store.go

```diff
@@ -163,50 +189,39 @@ func createImageIntegration(registry string, dce config.DockerConfigEntry, secur
 				Endpoint: registry,
 				Username: dce.Username,
 				Password: dce.Password,
-				Insecure: !secure,
```

Comment:
Correct.  The default for `Insecure` will be `false` (so we assume secure by default, only changed if the TLS check says otherwise during lazy init).  

Separately in the future (ROX-19663) we should consider implications of removing the TLS check all together and instead rely on the [image config CR](https://docs.openshift.com/container-platform/4.16/rest_api/config_apis/image-config-openshift-io-v1.html#image-config-openshift-io-v1:~:text=or%20imagestream%20imports.-,insecureRegistries,-array%20(string)) to determine which registries are 'insecure'. This may require some magic to obtain the CA certs that OCP uses to validate TLS, but nonetheless will be better from risk/security perspective.